Give players the option to have a custom title,

Addalyn opened the pull request:
custom titles will be handled by launcher
custom titles are only updated on game launch
todo mayby add timer?!? to update them
Also how are we going to handle versions?

Addalyn committed:
Give players the option to have a custom title,
custom titles will be handled by launcher
custom titles are only updated on game launch
todo mayby add timer?!? to update them
Also how are we going to handle versions?

Addalyn committed:
Whitespace

Addalyn committed:
Cleanup not used

Addalyn committed:
Move url to HydroConfig

Addalyn committed:
Fix

Zheneq committed:
Revert whitespace changes

Zheneq commented on Assembly-CSharp/PlayerTitleManager.cs:
```diff
@@ -0,0 +1,69 @@
+            return returnOnEmptyOverride;
+        }
+
+        foreach (var kvp in cachedTitlesByHandle)
+        {
+            string normalizedKey = kvp.Key.Split('#')[0].Trim();
+            if (normalizedKey == normalizedHandle)
+            {
+                foreach (PlayerTitle title in kvp.Value)
+                {
+                    string normalizedTitleHandle = title.Handle.Split('#')[0].Trim();
+                    if (normalizedTitleHandle == normalizedHandle)
+                    {
+                        return title.TitleText;
+                    }
+                }
+                break;
+            }
+        }
+
```
You don't need to iterate over a dictionary to get a value by key

Also, every player has multiple custom titles for different handles?

If keys have to normalized, it should happen during fetching (or they could be in ready-to-use state in the first place)

Zheneq commented on Assembly-CSharp/GameBalanceVars.cs:
```diff
@@ -1124,7 +1124,8 @@ public int CharacterExperienceToLevel(int currentLevel)
     }
 
-    public string GetTitle(int titleID, string returnOnEmptyOverride = "", int titleLevel = -1)
+#if VANILLA && SERVER
+    public string GetTitle(int titleID, string handle, string returnOnEmptyOverride = "", int titleLevel = -1, string handle)
```
```suggestion
#if VANILLA || SERVER
    public string GetTitle(int titleID, string returnOnEmptyOverride = "", int titleLevel = -1)
```

Addalyn committed:
Fix and add code suggestions
Also deleted PlayerTitle was useless and other things not used
Add refreshTitles run every 5min
and hopefully without whitespace changes

Addalyn committed:
put that back

Zheneq commented on Assembly-CSharp/GameBalanceVars.cs:
```diff
@@ -1124,7 +1124,8 @@ public int CharacterExperienceToLevel(int currentLevel)
     }
 
-    public string GetTitle(int titleID, string returnOnEmptyOverride = "", int titleLevel = -1)
+#if VANILLA && SERVER
+    public string GetTitle(int titleID, string handle, string returnOnEmptyOverride = "", int titleLevel = -1)
```
```suggestion
#if VANILLA || SERVER
    public string GetTitle(int titleID, string returnOnEmptyOverride = "", int titleLevel = -1)
```

Addalyn committed:
Update Assembly-CSharp/GameBalanceVars.cs

Co-authored-by: Zhenёq <Zheneq@users.noreply.github.com>
